pydantic_ai_slim/pydantic_ai/profiles/deepseek.py

```diff
@@ -1,8 +1,13 @@
 from __future__ import annotations as _annotations
 
-from . import ModelProfile
+from .openai import OpenAIModelProfile
 
 
-def deepseek_model_profile(model_name: str) -> ModelProfile | None:
+def deepseek_model_profile(model_name: str) -> OpenAIModelProfile | None:
     """Get the model profile for a DeepSeek model."""
-    return ModelProfile(ignore_streamed_leading_whitespace='r1' in model_name)
+    return OpenAIModelProfile(
+        ignore_streamed_leading_whitespace='r1' in model_name,
+        openai_chat_custom_reasoning_field='reasoning_content',
+        # For compatibility with existing behavior. May want to change later.
+        openai_chat_include_reasoning_in_request='combined',
```

Comment:
In the case of open source models, this is really a quality of the API/provider rather than the specific model, isn't it? I think I'd rather set these fields added in the `Provider.model_profile` methods, for those providers we know support a particular field.

Then I'm also OK changing the default behavior, because we're pretty sure that if someone's using that provider, they're using an API that will behave better when we use `separated`

For example `OllamaProvider` should be updated with the `reasoning` field.